lib/auth/auth.go

```diff
@@ -6854,18 +6855,21 @@ func (a *Server) getAccessRequestMonthlyUsage(ctx context.Context) (int, error)
 // If so, it returns an error. This is only applicable on usage-based billing plans.
 func (a *Server) verifyAccessRequestMonthlyLimit(ctx context.Context) error {
 	f := modules.GetModules().Features()
-	if f.IsLegacy() || f.IGSEnabled() {
-		return nil // unlimited
+	accessRequestsEntitlement := f.GetEntitlement(entitlements.AccessRequests)
+
+	if accessRequestsEntitlement.Limit == 0 {
```

Comment:
i just want us to give this a second thought: https://github.com/gravitational/teleport.e/pull/4317#discussion_r1643912368